# Daniil Litvinov


[![Linkedin Badge](https://img.shields.io/badge/-daniillitvinov-blue?style=flat-square&logo=Linkedin&logoColor=white&link=[https://www.linkedin.com/in/tanejasaksham/](https://www.linkedin.com/in/daniil-litvinov-b555b124b/))](https://www.linkedin.com/in/daniil-litvinov-b555b124b/) [![Website Badge](https://img.shields.io/badge/-daniillitvinov-03a57a?style=flat-square&labelColor=&logo=Elastic&link=https://danon6868.github.io/)](https://danon6868.github.io/) [![GitHub followers](https://img.shields.io/github/followers/danon6868?label=Follow&style=social)](https://github.com/danon6868/?tab=follow) [![Gmail Badge](https://img.shields.io/badge/-daniillitvinov997@gmail.com-c14438?style=flat-square&logo=Gmail&logoColor=white&link=mailto:daniillitvinov997@gmail.com)](mailto:daniillitvinov997@gmail.com) [![Telegram Badge](https://img.shields.io/badge/-@Danil_litvinov-blue?style=flat-square&logo=Telegram&logoColor=white&link=[https://www.linkedin.com/in/tanejasaksham/](https://t.me/Danil_litvinov))](https://t.me/Danil_litvinov)

I am a computational biologist with a commercial experience in python and R programming and in machine learning. My industry experience is applying machine learning methods and statistics for analyzing omics data and computer vision methods for cryo-EM data analysis.

In addition, I have experience teaching statistics and machine learning for students of biology and medicine. My primary goal is to continue to develop my skills in the field of AI/ML/DL applied to biology.

## Key projects

### LoReNNMap

[Application](https://www.bioeng.ru/lorennmap/). [Source](https://github.com/danon6868/CryoEM_Resolution_Estimation)

Developed an algorithm for cryo-EM maps resolution estimation using deep learning.

  * Collected and prepared training data, pairs of electron density maps, and local resolution maps.
  * Developed and trained neural network model to estimate local resolution map based on electron density map using 3D‑UNet model architecture.
  * Based on the trained model, created a CLI tool as well as a web application for cryo‑EM maps resolution estimation using deep learning.
  * **Skills:** Python, Django, PyTorch, Keras, EMAN2, RELION-3, scikit-learn, Bash.

### Portfolio website

[Application](https://danon6868.github.io/). [Source](https://github.com/danon6868/portfolio-website)

Python Django-based personal portfolio website. All content: personal information, portfolio projects, social media links, etc. can be adjusted in Wagtail admin.

  * Created a Django‑based personal portfolio website.
  * The website uses Wagtail CMS. Wagtail is a Django Content Management System.
  * Code from the attached repository can be used as a starting point for developing a production‑ready Django personal website with deployment to Heroku.
  * **Skills:** Python, Docker, JavaScript, CSS, HTML, Django, Wagtail, SQL.

### Recommender system

[Results](https://github.com/danon6868/recommender-system-api/blob/main/recommender_system/ab_test_results_analysis/ab_test_result_analysis.ipynb). [Source](https://github.com/danon6868/recommender-system-api)

Content-based recommender system API based on the text of the post and user data.

  * Created a content‑based recommendation system based on user data and post texts.
  * Built a web API using FastAPI to predict the most relevant posts for a user.
  * Added an A/B testing system to select the most appropriate recommendation model.
  * **Skills:** Python, Docker, SQL, PyTorch, Scikit-learn, CatBoost, FastAPI, Optuna, NLP, Transformers.

### Developing best practices for single-cell analysis: data integration

[Results](https://docs.google.com/presentation/d/1yZ4Zv2htTKizPDkxDkMDWy77uvzFRmiguy3UsyZDXGk/edit#slide=id.gdd663730a8_0_0). [Source](https://github.com/immunomind/bi2021spring)

The single-cell transcriptomics analysis has multiple steps, but we have focused on data integration — a crucial step when working with clinical data coming from patients.

  * Researched libraries for scRNA‑seq data analysis and batch effect removal.
  * Tested 8 algorithms on different data types and select the most suitable one for each case.
  * Created software for batch effect removal from scRNA‑seq data using the best algorithm for the current case.
  * **Skills:** Python, R, Scikit-learn, Scanpy, BBKNN, MNN, Scanorama, Cell Ranger, Bash.

### Study of the transcriptome during intense exercises in highlands

[Results](https://docs.google.com/presentation/d/1EzHlLa3YpSoFgdo5eapeyzFIxL91gq_KU4ZYvp_kF3k/edit#slide=id.p). [Source](https://github.com/danon6868/sky_runners)

This project aims to study differential genes expression of 19 sportsmen during physical and psychological stress before and after running in extreme highlands conditions.

  * Processed and evaluated the quality of raw reads.
  * Aligned processed reads on the human reference genome and counted gene and isoform expression levels.
  * Identify the list of differential expressed genes and their functional analysis as well as cluster analysis.
  * **Skills:** Python, R, DESeq2, FastQC, Bash, STAR, RSEM, MSigDB, GeneQuery.

<details>
<summary>
My skills :hammer_and_wrench:
</summary>
  
### Programming

  * **Python** (Numpy, Pandas, Matplotlib, Seaborn, Sklearn, PyTorch, Keras, FastAPI, Django)
  * **R** (ggplot2, Seurat, DeSeq2, dplyr)
  * **Linux**, Bash, **git**, GitHub, Bitbucket, **Docker**, Kubernetes, Airflow, **Jira**
  * **JavaScript**
  
### Machine Learning Methods
  
  * **Classical Machine Learning** (linear models, tree-based approaches, Catboost, LightGBM, XGBoost, Bayesian methods)
  * **Deep learning** (MLP, CNN, image segmentation, detection, RNN, LSTM, Transformers, AE, VAE, GAN, TabNet)
  * **Model tuning** (Optuna, genetic algorithm, Boruta)
  * **Interpretable machine learning** (SHAP, LIME, Pixel Attribution)

### Statistics

  * **Hypothesis testing**, ANOVA
  * **Survival analysis** (Kaplan-Meier curves, Log rank test, Cox regression)
  * **Casual inference** (Propensity score, Instrumental variables, Regression discontinuity)
  
### Bioinformatics
  
  * **Databases** (NCBI, UniProt, PDB, MsigDB, SILVA)
  * **Command-line tools** (Cellranger, cellSNP, Picard, BLAST, GATK, STAR, SPAdes)
  * **Protein sequence analysis tools** (MAFFT, MUSCLE, HMMER, ESM)
  * **Protein structure analysis tools** (Rosetta, Phenix, Coot, AlphaFold)
  
### Languages
  
  * **Russian** – Native
  * **English** – Full professional proficiency
  * **German** – Elementary proficiency
  
### Life Sciences
  
  * **Biological education**, which helps me to understand specialized biological and medical literature
  * Work experience in **molecular**, **microbiological**, and **biochemical** labs
  
### Soft skills
  
  * **Agile** software development methods
  * Presentation skills
  
</details>

-------

[![GitHub Streak](http://github-readme-streak-stats.herokuapp.com?user=danon6868&theme=dark&background=000000)](https://git.io/streak-stats)
