# Daniil Litvinov


[![Linkedin Badge](https://img.shields.io/badge/-daniillitvinov-blue?style=flat-square&logo=Linkedin&logoColor=white&link=[https://www.linkedin.com/in/tanejasaksham/](https://www.linkedin.com/in/daniil-litvinov-b555b124b/))](https://www.linkedin.com/in/daniil-litvinov-b555b124b/) [![Website Badge](https://img.shields.io/badge/-daniillitvinov-03a57a?style=flat-square&labelColor=&logo=Elastic&link=https://danon6868.github.io/)](https://danon6868.github.io/) [![GitHub followers](https://img.shields.io/github/followers/danon6868?label=Follow&style=social)](https://github.com/danon6868/?tab=follow) [![Gmail Badge](https://img.shields.io/badge/-daniillitvinov997@gmail.com-c14438?style=flat-square&logo=Gmail&logoColor=white&link=mailto:daniillitvinov997@gmail.com)](mailto:daniillitvinov997@gmail.com) [![Telegram Badge](https://img.shields.io/badge/-@Danil_litvinov-blue?style=flat-square&logo=Telegram&logoColor=white&link=[https://www.linkedin.com/in/tanejasaksham/](https://t.me/Danil_litvinov))](https://t.me/Danil_litvinov)

I am a computational biologist with a commercial experience in python and R programming and in machine learning. My industry experience is applying machine learning methods and statistics for analyzing omics data and computer vision methods for cryo-EM data analysis.

In addition, I have experience teaching statistics and machine learning for students of biology and medicine. My primary goal is to continue to develop my skills in the field of AI/ML/DL applied to biology.

## Key projects

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

### LoReNNMap [<i class="fa fa-github" style="color:white"></i>](https://github.com/danon6868/CryoEM_Resolution_Estimation) [<i class="fa fa-web" style="color:white"></i>](https://www.bioeng.ru/lorennmap/) 

Developed an algorithm for cryo-EM maps resolution estimation using deep learning.

  * Collected and prepared training data, pairs of electron density map and local resolution map.
  * Developed and trained neural network model that can estimate local resolution map based on electron density map using 3D‑UNet model architecture.
  * Based on the trained model, created CLI tool as well as web application for cryo‑EM maps resolution estimation using deep learning.
  * **Technical Skills:** Python, Django, PyTorch, Keras, EMAN2, RELION‑3, Scikit‑learn, Bash.

### Recommender system [<i class="fa fa-github" style="color:white"></i>](https://github.com/danon6868/recommender-system-api)

Content-based recommender system API based on the text of the post and user data.

  * Created content‑based recommendation system based on user data and post texts.
  * Built a web API using FastAPI to predict the most relevant posts for a user.
  * Added an A/B testing system to select the most appropriate recommendation model.
  * **Technical Skills:** Python, Docker, SQL, PyTorch, Scikit‑learn, CatBoost, FastAPI, Optuna, NLP, Transformers.

<details>
<summary>
My skills :hammer_and_wrench:
</summary>
  
### Programming

  * **Python** (Numpy, Pandas, Matplotlib, Seaborn, Sklearn, PyTorch, Keras, FastAPI, Django)
  * **R** (ggplot2, Seurat, DeSeq2, dplyr)
  * **Linux**, **Bash**, **git**, **Docker**, **Kubernetes**, **Airflow**
  * **JavaScript**
  
### Machine Learning Methods
  
  * **Classical Machine Learning** (linear models, tree-based approaches, Catboost, LightGBM, XGBoost, Bayesian methods)
  * **Deep learning** (MLP, CNN, image segmentation, detection, RNN, LSTM, Transformers, AE, VAE, GAN, TabNet)
  * **Model tuning** (Optuna, genetic algorithm, Boruta)
  * **Interpretable machine learning** (SHAP, LIME, Pixel Attribution)

<!-- ### Statistics

  * 
  *
  * -->
  
### Bioinformatics
  
  * **Databases** (NCBI, UniProt, PDB, MsigDB, SILVA)
  * **Command-line tools** (Cellranger, cellSNP, Picard, BLAST, GATK, STAR, SPAdes)
  * **Protein sequence analysis tools** (MAFFT, MUSCLE, HMMER, ESM)
  * **Protein structure analysis tools** (Rosetta, Phenix, Coot, AlphaFold)
  
### Languages
  
  * **Russian** – Native
  * **English** – Full professional proficiency
  * **German** – Elementary proficiency
  
### Life Sciences
  
  * **Biological education**, which helps me to understand specialized biological and medical literature
  * Work experience in **molecular**, **microbiological**, and **biochemical** labs
  
### Soft skills
  
  * **Agile** software development methods
  * Presentation skills
  
</details>

-------

[![GitHub Streak](http://github-readme-streak-stats.herokuapp.com?user=danon6868&theme=dark&background=000000)](https://git.io/streak-stats)
